### ISSUE TITLE
Improved feature parity: Stats profiles for virtuals

### ISSUE THREAD

**Issue by jussisjostrom** [F5Networks/f5-appsvcs-extension#526]
### Is your feature request related to a problem? Please describe.
I ran into this missing functionality in AS3, when porting a viprion configuration to be applied using AS3 on a VNF, ie pile of VE's. The virtuals use a custom statistics counter/metric giving stats on certain type of traffic that is defined as a key performance indicator of this system. 

The specific metric of interest is defined in a statistics profile:
```
ltm profile statistics /Common/myCustomStatsProfile {
    defaults-from /Common/stats
    field1 theCustomMetric
}
```
This statistics profile is attached to virtual server:
```
ltm virtual /Common/exampleVirtual {
...
    profiles {
         /Common/myCustomStatsProfile
         ...
```
An irule that is attached to the virtual server for the traffic of interest  increases theCustomMetric counter using STATS::incr command. Then the metric is exposed via SNMP MIB in the  ltmUserStatProfileStatTable oid and it is polled using external SNMP monitoring system. Currently I cannot create this configuration using AS3.

### Describe the solution you'd like
I would appreciate a feature parity in AS3 compared to tmsh, to support creation of statistics profile (ltm profile statistics) and configure the statistics profile for a virtual server.

### Describe alternatives you've considered
Currently, I will need to create the statistics profile outside AS3 and affect the statistics profile to the virtuals after AS3 declaration has been posted. This makes it quite cumbersome to apply on a large scale VNF.

### Additional context
The desired tmsh configuration is shown above.

**Comment by LaurentGravier** [comment]
Hi,

I face exactly the same problem. On top of having to configure the statistics profiles out of AS3 (and attaching them to the Virtual Servers after the VS are created with AS3), this makes it impossible to configure the Virtual Servers with the iRules in AS3 (if the iRule references a statistics profile which is not attached to the VS, the declaration will fail). 

So I have to create the Virtual servers in AS3, then, with non declarative commands, attach the statistics profiles and the iRules to the Virtual Servers.

This cancels a big part of the benefits of AS3 (re-sending the AS3 declaration will remove the iRules and the statistics profiles from the VS...). 

Thanks in advance !

Laurent Gravier.

**Comment by sunitharonan** [comment]
This has been resolved in the 3.41.0 release of AS3.
